scripts/pre_commit.sh

```diff
@@ -1,15 +1,16 @@
 #!/bin/bash
 set -e
 
-./scripts/build_proto.sh
+# start tests in parallel (with additional parallelism)
+pytest -m fast -n auto &
+
+# check bandit in parallel
+bandit -r src -ll &
+
+# run API documentation test notebooks in parallel
+./scripts/nb_test.sh && pytest examples/api --cov-fail-under 0 &
 
 # fix isort and format with black
-isort .
-black src tests
-pre-commit run --all-files
-bandit -r src -ll
-pytest -m fast
-
-# run API documentation test notebooks
-./scripts/nb_test.sh
-pytest examples/api --cov-fail-under 0
\ No newline at end of file
+./scripts/build_proto.sh && isort . && black src tests && pre-commit run --all-files &
+
+wait
```

Comment:
```  wait [n ...]
        Wait  for each specified process and return its termination sta-
        tus.  Each n may be a process ID or a job  specification;  if  a
        job  spec  is  given,  all  processes in that job's pipeline are
        waited for.  If n is not given, all currently active child  pro-
        cesses  are  waited  for,  and  the return status is zero.  If n
        specifies a non-existent process or job, the  return  status  is
        127.   Otherwise,  the  return  status is the exit status of the
        last process or job waited for.```

Taken from [here](https://stackoverflow.com/questions/13296863/difference-between-wait-and-sleep).
It specifies that using wait without any process, the exist status would be 0.
Maybe we should record each pid and at the end wait in 'reverse' order on how much we expected them to run?

Reply:
You are right, looks like i'm using it wrong. I have removed this for now to avoid confusion that it does anything.